crates/type-system/src/ontology/mod.rs

```diff
@@ -6,10 +6,43 @@
 //! aid with the de/serialization, intermediary structs and helpers are defined across various
 //! submodules.
 
-pub mod data_type;
-pub mod property_type;
+mod data_type;
+mod property_type;
 // TODO: reconsider calling these URIs in the spec, it seems to be a redundant term nowadays and
 //  we should probably just go with URL
 pub mod uri;
 
-pub mod repr_shared;
+mod shared;
+
+pub use data_type::{DataType, DataTypeReference, ParseDataTypeError};
+pub use property_type::{
+    ParsePropertyTypeError, PropertyType, PropertyTypeReference, PropertyValues,
+};
+pub use shared::{
+    array::{
+        error::{ParseOneOfArrayError, ParsePropertyTypeReferenceArrayError},
+        Array, ValueOrArray,
+    },
+    object::{error::ParsePropertyTypeObjectError, Object},
+    one_of::{error::ParseOneOfError, OneOf},
+    validate::{ValidateUri, ValidationError},
+};
+
+// Re-export the repr contents so they're nicely grouped and so that they're easier to import in
+// a non-ambiguous way where they don't get confused with their non repr counterparts.
+// For example, `import crate::ontology::repr` lets you then use `repr::DataType`
+#[expect(
+    clippy::redundant_pub_crate,
```

Comment:
We replaced this lint in `error-stack` with `unreachable_pub`

Reply:
Have done so, although it behaved a little weirdly.